### PR TITLE
Implement psutil within process_exists function

### PR DESCRIPTION
**Pull request by lleshchi**
Related stratis-storage/stratis-cli#392
Needs to be tested by Bryan

Signed-off-by: Leah Leshchinsky <lleshchi@redhat.com>

### CODE REVIEW

File: tests/blackbox/testlib/utils.py [hunk 1]
```diff
@@ -53,16 +56,19 @@ def random_string(length=4):
 
 def process_exists(name):
     """
-    Walk the process table looking for executable 'name', returns pid if one
-    found, else return None
+    Look through processes, using their pids, to find one matching 'name'.
+    Return None if no such process found, else return the pid.
+    :param name: name of process to check
+    :type name: string
+    :return: None or pid
```
Comment (mulkieran) [inline tests/blackbox/testlib/utils.py]: You can also use the ```rtype``` markup, which should say something like: ```NoneType or int```

Comment (lleshchi) [reply tests/blackbox/testlib/utils.py]: Is this in addition to `return`, or instead of it?

Comment (mulkieran) [reply tests/blackbox/testlib/utils.py]: In addition: ```:param``` is to ```:type``` as ```:return``` is to ```:rtype".

**Comment by lleshchi** [comment]
May `process_iter` be extended to the whitebox tests as well?

**Comment by bgurney-rh** [comment]
I was able to successfully test this; however, I had to manually install python3-psutil to get it working.

I created a CI PR to have this package installed by the stratis-blackbox-run.sh script.

**Comment by bgurney-rh** [comment]
The CI PR that installs python3-psutil as part of running stratis-blackbox-run.sh has been merged, so that part should be all set.

File: tests/blackbox/testlib/utils.py [hunk 2]
```diff
@@ -53,16 +56,20 @@ def random_string(length=4):
 
 def process_exists(name):
     """
-    Walk the process table looking for executable 'name', returns pid if one
-    found, else return None
+    Look through processes, using their pids, to find one matching 'name'.
+    Return None if no such process found, else return the pid.
+    :param name: name of process to check
+    :type name: string
```
Comment (jbaublitz) [inline tests/blackbox/testlib/utils.py]: @mulkieran do we want this to be `str`?

Comment (mulkieran) [reply tests/blackbox/testlib/utils.py]: > @mulkieran do we want this to be `str`?

Yes, it should be ```str```.

File: tests/blackbox/testlib/utils.py [hunk 3]
```diff
@@ -53,16 +56,20 @@ def random_string(length=4):
...
+    :return: pid or None
+    :rtype: int or NoneType
     """
-    for pid in [pid for pid in os.listdir("/proc") if pid.isdigit()]:
+    for proc in psutil.process_iter():
```
Comment (mulkieran) [inline tests/blackbox/testlib/utils.py]: If you do use ```process_iter``` you might as well get the benefit. I think there is some design where you can specify that it should just get the name, and not all the information about the process, which is far more efficient. You might as well try that out.

**Comment by mulkieran** [comment]
> May `process_iter` be extended to the whitebox tests as well?

That would be fine, but we'ld file an issue and do it in a separate PR, just because the blackbox tests are so special.

**Pull request closed without merging**